### PR TITLE
Maint/add ci

### PR DESCRIPTION
**Pull request by JWhitleyAStuff**
This removes the Gitlab and Travis config files for CI and adds a CircleCI config file. I did tons of testing (see the comments on the first commit in this PR) and found that 1 thread per `make` instance and 3 workers per call to `colcon` is the max that CircleCI can handle without the compiler running out of memory. Since, even at this level, the build takes approximately 25 minutes, I recommend we turn on the CircleCI option to only build PRs and the primary branch (for us, `as/master`).

### CODE REVIEW

**Comment by JWhitleyAStuff** [comment]
Successful build: https://circleci.com/gh/astuff/autoware/21

I would still wait for the current build to complete just to make sure.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            pip3 install -U setuptools
+            apt install -y python3-colcon-common-extensions
+      - run:
+          name: Install ROS Dependencies
+          command: |
+            source `find /opt/ros -name setup.bash | sort | head -1`
+            cd ros/src
+            wstool init
+            test -f "dependencies.rosinstall" && wstool merge "dependencies.rosinstall"
+            wstool up
```
Comment (icolwell-as) [inline .circleci/config.yml]: I'm not familiar with wstool, but why is it needed?
I would have assumed a regular rosdep update and install as directed in the [install docs](https://github.com/autowarefoundation/autoware/wiki/Source-Build) would be fine.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+version: 2
+jobs:
+  kinetic:
+    docker:
+      - image: autonomoustuff/docker-builds:kinetic-perception
```
Comment (icolwell-as) [inline .circleci/config.yml]: If we want to shave off a couple minutes from the build, we could create an image that already contains all the autoware deps, then rosdep will just double-check and move on.

Comment (JWhitleyAStuff) [reply .circleci/config.yml]: This is true. I had thought about it as well. Would you mind creating a new Dockerfile in https://github.com/astuff/docker-builds in the ros/kinetic/ folder? I'll set up the Docker Hub automated build after that.

Comment (icolwell-as) [reply .circleci/config.yml]: Will do!

Comment (JWhitleyAStuff) [reply .circleci/config.yml]: @icolwell-as - Any movement on the Docker container?

Comment (JWhitleyAStuff) [reply .circleci/config.yml]: She's a-build-n'. https://cloud.docker.com/u/autonomoustuff/repository/docker/autonomoustuff/docker-builds/timeline

**Pull request merged**